app/screens/post_options/components/reaction_bar/reaction_bar.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {useWindowDimensions, View} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {Screens} from '@constants';
+import {
+    LARGE_CONTAINER_SIZE,
+    LARGE_ICON_SIZE,
+    REACTION_PICKER_HEIGHT,
+    SMALL_CONTAINER_SIZE,
+    SMALL_ICON_BREAKPOINT,
+    SMALL_ICON_SIZE,
+} from '@constants/reaction_picker';
+import {useIsTablet} from '@hooks/device';
+import {showModal} from '@screens/navigation';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+import PickReaction from './components/pick_reaction';
+import Reaction from './components/reaction';
+import {ReactionProvider} from './components/reaction_context';
+
+type QuickReactionProps = {
+    theme: Theme;
+    recentEmojis: string[];
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            backgroundColor: theme.centerChannelBg,
+            flexDirection: 'row',
+            alignItems: 'center',
+            height: REACTION_PICKER_HEIGHT,
+            justifyContent: 'space-between',
+            marginTop: -16,
+        },
+    };
+});
+
+const ReactionBar = ({recentEmojis = [], theme}: QuickReactionProps) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const {width} = useWindowDimensions();
+    const [selectedEmoji, setSelectedEmoji] = useState('');
+    const isSmallDevice = width < SMALL_ICON_BREAKPOINT;
+    const styles = getStyleSheet(theme);
+
+    const handleEmojiPress = useCallback((emoji: string) => {
+        setSelectedEmoji(emoji);
+        // eslint-disable-next-line no-console
+        console.log('>>>  selected this emoji', emoji);
+    }, []);
+
+    const openEmojiPicker = useCallback(async () => {
+        // DeviceEventEmitter.emit(Navigation.NAVIGATION_CLOSE_MODAL);
```

Comment:
There is a strange behaviour when opening the picker on the Tablet.. We'll need to review